### PR TITLE
[LLM] Llama 3.1 serving example

### PR DESCRIPTION
**Pull request by romilbhardwaj**
Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: docs/source/_gallery_original/index.rst [hunk 1]
```diff
@@ -39,6 +39,7 @@ Contents
    DBRX (Databricks) <llms/dbrx>
    Llama-2 (Meta) <llms/llama-2>
    Llama-3 (Meta) <llms/llama-3>
+   Llama-3.1 (Meta) <llms/llama-3_1>
```
Comment (concretevitamin) [inline docs/source/_gallery_original/index.rst:42]: We should add to docs home page's LLM list too.

File: docs/source/_static/custom.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ document.addEventListener('DOMContentLoaded', () => {
         { selector: '.toctree-l1 > a', text: 'Running on Kubernetes' },
         { selector: '.toctree-l1 > a', text: 'Ollama' },
         { selector: '.toctree-l1 > a', text: 'Llama-3 (Meta)' },
-        { selector: '.toctree-l1 > a', text: 'Qwen (Alibaba)' },
+        { selector: '.toctree-l1 > a', text: 'Llama-3.1 (Meta)' },
```
Comment (concretevitamin) [inline docs/source/_static/custom.js:33]: Can remove L31/32 now?

File: llm/llama-3_1/README.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
+# Serve Llama 3.1 on Your Own Infrastructure
+
+![](https://i.imgur.com/kQGzHI6.png)
```
Comment (concretevitamin) [inline llm/llama-3_1/README.md:3]: Probably want this to have smaller width. In README preview it's quite wide.

File: llm/llama-3_1/README.md [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+
+```bash
+sky check
+```
+
+See [docs](https://skypilot.readthedocs.io/en/latest/getting-started/installation.html) for details.
+
+### Step 1: Get a GPU dev node (pod or VM)
+
+> **Tip:** Skip directly to Step 3 to get the SkyPilot YAML to deploy Llama 3.1 on your infra.
```
Comment (concretevitamin) [inline llm/llama-3_1/README.md:54]: Wdym? I guess this was supposed to say "If you simply want the final deployment YAML, ..."?

Comment (concretevitamin) [reply llm/llama-3_1/README.md:54]: Can also link to Step 3.

File: llm/llama-3_1/README.md [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+  - Meta-Llama-3.1-405B-FP8
+- Instruction tuned:
+  - Meta-Llama-3.1-8B-Instruct
+  - Meta-Llama-3.1-70B-Instruct
+  - Meta-Llama-3.1-405B-Instruct-FP8
+
+
+The full precision 405B model Meta-Llama-3.1-405B requires multi-node inference and is work in progress - join the [SkyPilot community Slack](https://slack.skypilot.co/) for discussions.
+
+If you are wondering what GPU you need for each model, here is the compatibility matrix (applies to both pretrained and instruction tuned models):
```
Comment (concretevitamin) [inline llm/llama-3_1/README.md:104]: This table is pretty useful. How about putting it into its own section before Step 0? e.g., "What hardware to use for each Llama 3.1 model"?

**Review by concretevitamin — COMMENTED** [review]
Awesome @romilbhardwaj, some quick comments.

- [ ] add to docs & repo README's LLM list too, for both finetuning & serving?

**Comment by romilbhardwaj** [comment]
Thanks, resolved comments.

**Review by concretevitamin — APPROVED** [review]

File: llm/llama-3_1/README.md [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+
+- Local GPU workstation
+- Kubernetes cluster
+- Cloud accounts ([12 clouds supported](https://skypilot.readthedocs.io/en/latest/getting-started/installation.html))
+
+SkyPilot will be used as the unified framework to launch serving on any (or multiple) infra that you bring.
+
+## Serving Llama 3.1 on your infra
+
+We will first test the model on a GPU dev node, then package it for deployment using SkyPilot.
```
Comment (Michaelvll) [inline llm/llama-3_1/README.md:22]: Can we have highlighted sentence for the complete deployment YAML so people can directly jump to that section? 
```suggestion
We offer a step-by-step guide for how to use SkyPilot to test a new model on a GPU dev node, and then package it for deployment using SkyPilot. **For the complete deployment guide of Llama 3.1, see [Step 3: Package and deploy using SkyPilot](#Step-3-Package-and-deploy-using-SkyPilot)
```

**Pull request merged**